File: App/Models/IntentCollection.swift
import Foundation

/// A shooting-intent collection — the top-level way users browse poses.
/// This is the app's answer to "what are you shooting today?", and the primary
/// structural difference from a mode-first camera app.
enum IntentCollection: String, CaseIterable, Identifiable, Hashable {
    case dating, professional, mirror, fullbody, couple, candid

    var id: String { rawValue }

    var title: String {
        switch self {
        case .dating:       return "dating & profile"
        case .professional: return "professional"
        case .mirror:       return "mirror selfie"
        case .fullbody:     return "full body"
        case .couple:       return "couples"
        case .candid:       return "candid"
        }
    }

    var subtitle: String {
        switch self {
        case .dating:       return "shots that spark a swipe right"
        case .professional: return "headshots that mean business"
        case .mirror:       return "the effortless mirror moment"
        case .fullbody:     return "head-to-toe, framed right"
        case .couple:       return "two people, one great frame"
        case .candid:       return "caught-in-the-moment, on purpose"
        }
    }

    var systemImage: String {
        switch self {
        case .dating:       return "sparkles"
        case .professional: return "briefcase"
        case .mirror:       return "rectangle.portrait"
        case .fullbody:     return "figure.stand"
        case .couple:       return "figure.2"
        case .candid:       return "camera.viewfinder"
        }
    }

    /// No poses shipped yet — shown but not enterable. Couples poses arrive in Phase 2.
    var comingSoon: Bool { self == .couple }
}

File: android/app/src/main/kotlin/com/oerol/pose/data/IntentCollection.kt
package com.oerol.pose.data

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.AutoAwesome
import androidx.compose.material.icons.outlined.CameraAlt
import androidx.compose.material.icons.outlined.CropPortrait
import androidx.compose.material.icons.outlined.Groups
import androidx.compose.material.icons.outlined.PersonOutline
import androidx.compose.material.icons.outlined.WorkOutline
import androidx.compose.ui.graphics.vector.ImageVector

/**
 * A shooting-intent collection — the top-level way users browse poses.
 * Mirrors iOS App/Models/IntentCollection.swift; ids/titles/subtitles must match exactly.
 */
enum class IntentCollection(
    val id: String,
    val title: String,
    val subtitle: String,
    val icon: ImageVector,
    val comingSoon: Boolean = false,
) {
    DATING("dating", "dating & profile", "shots that spark a swipe right", Icons.Outlined.AutoAwesome),
    PROFESSIONAL("professional", "professional", "headshots that mean business", Icons.Outlined.WorkOutline),
    MIRROR("mirror", "mirror selfie", "the effortless mirror moment", Icons.Outlined.CropPortrait),
    FULLBODY("fullbody", "full body", "head-to-toe, framed right", Icons.Outlined.PersonOutline),
    COUPLE("couple", "couples", "two people, one great frame", Icons.Outlined.Groups, comingSoon = true),
    CANDID("candid", "candid", "caught-in-the-moment, on purpose", Icons.Outlined.CameraAlt),
}
